### DISCUSSION TITLE
Can I create worker nodes in Docker to temporarily improve my resource pool for an rpi cluster?

### DISCUSSION THREAD

**Discussion by devantler** [siderolabs/talos#7827] [Q&A] [answered]
Hey all,

I am building my Homelab with [Talos and Flux GitOps](https://github.com/devantler/homelab), on a three-node RPI cluster. I know I will run out of resources at some point, so I am investigating whether creating Talos worker nodes in Docker can alleviate some of the load on the RPI cluster to my main machine until I get more machinery in the home.

I have not found documentation covering my use case, so I thought I wanted to ask here ✌🏻

**Comment by smira** [comment, marked as answer]
yes, this should work: https://www.talos.dev/v1.5/talos-guides/install/local-platforms/docker/#running-talos-in-docker-manually

**Reply by devantler** [reply]
Ah thank you! Totally missed that.

**Comment by ipclaudio** [comment]
@devantler @smira 
Please, can you write an example how to add a worker on an existing cluster using docker after running : 
`talosctl cluster create --wait`


```
 talosctl cluster  show
PROVISIONER       docker
NAME              talos-default
NETWORK NAME      talos-default
NETWORK CIDR      10.5.0.0/24
NETWORK GATEWAY
NETWORK MTU       1500

NODES:

NAME                           TYPE           IP         CPU   RAM   DISK
talos-default-controlplane-1   controlplane   10.5.0.2   -     -     -
talos-default-worker-1         worker         10.5.0.3   -     -     -
```

**Reply by smira** [reply]
`talosctl cluster create` doesn't support adding nodes to the cluster after creation.

you can try to add manually, but there is no "easy" way atm

**Reply by ipclaudio** [reply]
thanks @smira 

what is another  easier provisioner on ubuntu with the ability to add worker ?

**Reply by smira** [reply]
You can run Talos anywhere, follow the guide https://www.talos.dev/v1.6/introduction/getting-started/ and deploy it wherever you like. 

You can use libvirt and various frontends for it. You can even use Omni with libvirt VMs.

**Reply by ehcpdeveloper** [reply]
> `talosctl cluster create` doesn't support adding nodes to the cluster after creation.
> 
> you can try to add manually, but there is no "easy" way atm

1-Hi, is this info still correct?
2-howto add nodes to a Talos created with "talos cluster create"?
3-if no easy way, what is manual way?
4-Howto add Talos node (that is in Docker) onto this existing cluster, from a server at remote location.

**Reply by devantler** [reply]
I ended up not going down this route, and instead spin up Talos VMs with [UTM](https://mac.getutm.app), so sadly I can't share any "workarounds". I think I can answer your questions though.

1. I believe the info is correct. The docs have not changed, so you are on your own.
2. It is not possible to add nodes to a cluster created with "talos cluster create". Remember this was never intended for production clusters, but for running and testing out stuff locally (kinda like Kind or K3d).
3. You are on your own here, but there is some info to help you here: https://www.talos.dev/v1.5/talos-guides/install/local-platforms/docker/#running-talos-in-docker-manually
4. You would need access to the Docker socket, so maybe through SSH or a secure tunnel.
